### PR TITLE
Fix drupal_coder_exec bugs

### PR DESCRIPTION
**Pull request by wvu**
Fixes some things missed in #7115. My bad for not catching these before merging.
- [x] Test the `cmd/unix/generic` payload
- [x] Make sure `payload` isn't overwritten
- [x] Make sure the module cleans up after itself

cc @mmetince

### CODE REVIEW

**Comment by wwebb-r7** [comment]
### Release Notes

This merge corrects a few minor issues in exploits/unix/webapp/drupal_coder_exec - originally #7115

**Comment by mmetince** [comment]
Using ```on_new_session``` is a good idea, I didn't there was a function for that but it causes very weird messages on console. 

It will dissapear when I remove the on_new_session function declaration from module.

Any idea @wvu-r7 ? 

screen shot 2016-08-18 at 22 35 56

**Comment by wvu-r7** [comment]
@mmetince: That's ```shell_command_token``` working its magic. Those are just markers.

File: modules/exploits/unix/webapp/drupal_coder_exec.rb [hunk 1]
```diff
@@ -87,15 +87,23 @@ def exploit
...
       'uri' => normalize_uri(target_uri.path, 'sites/all/modules/coder/coder_upgrade/scripts/coder_upgrade.run.php'),
       'encode_params' => false,
       'vars_get' => {
-        'file' => payload
+        'file' => pl
       }
     )
   end
+
+  # XXX: FileDropper can't handle weird filenames
```
Comment (wchen-r7) [inline modules/exploits/unix/webapp/drupal_coder_exec.rb]: Huh? What is weird?

Comment (wvu-r7) [inline modules/exploits/unix/webapp/drupal_coder_exec.rb]: The payload *is* the filename. I might be able to fix the mixin with ```./```.

**Pull request merged**